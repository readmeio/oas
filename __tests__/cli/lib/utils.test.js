const utils = require('../../../src/cli/lib/utils');

describe('CLI utils', function () {
  it.todo('#findSwagger');

  describe('#isJSONOrYaml', function () {
    it('should recognize `.json` as JSON', () => {
      expect(utils.isJSONOrYaml(require.resolve('@readme/oas-examples/3.0/json/petstore.json'))).toBe(true);
    });

    it('should recognize `.yaml` as YAML', () => {
      expect(utils.isJSONOrYaml(require.resolve('@readme/oas-examples/3.0/yaml/petstore.yaml'))).toBe(true);
    });

    it('should recognize `.yml` as YAML', () => {
      expect(utils.isJSONOrYaml(require.resolve('@readme/oas-examples/3.0/yaml/petstore.yaml'))).toBe(true);
    });

    it('should not recognize `.js` as JSON', () => {
      expect(utils.isJSONOrYaml('i-am-not-json.js')).toBe(false);
    });

    it('should not recognize `.y` as YAML', () => {
      expect(utils.isJSONOrYaml('i-am-not-yaml.y')).toBe(false);
    });
  });

  it.todo('#fileExists');

  it.todo('#guessLanguage');

  describe('#swaggerInlineExample', () => {
    it.each([['coffee'], ['go'], ['java'], ['js'], ['jsx'], ['php'], ['py'], ['rb'], ['ts']])(
      'should support `.%s`',
      lang => {
        expect(utils.swaggerInlineExample(lang)).toMatchSnapshot();
      }
    );
  });
});
