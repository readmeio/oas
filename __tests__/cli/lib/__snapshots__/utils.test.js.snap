// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI utils #swaggerInlineExample should support \`.coffee\` 1`] = `
"  [36m###[39m
  [36m@oas [get] /pet/{petId}[39m
  [36mdescription: Returns all pets from the system that the user has access to[39m
  [36mparameters:[39m
  [36m  - (path) petId=2* {Integer} The pet ID[39m
  [36m  - (query) limit {Integer:int32} The number of resources to return[39m
  [36m###[39m
  [90mroute.get '/pet/:petId', pet.show[39m"
`;

exports[`CLI utils #swaggerInlineExample should support \`.go\` 1`] = `
"  [36m/*[39m
  [36m * [39m[36m@oas [get] /pet/{petId}[39m
  [36m * [39m[36mdescription: Returns all pets from the system that the user has access to[39m
  [36m * [39m[36mparameters:[39m
  [36m * [39m[36m  - (path) petId=2* {Integer} The pet ID[39m
  [36m * [39m[36m  - (query) limit {Integer:int32} The number of resources to return[39m
  [36m */[39m
  [90mfunc getPet(id) {[39m"
`;

exports[`CLI utils #swaggerInlineExample should support \`.java\` 1`] = `
"  [36m/*[39m
  [36m * [39m[36m@oas [get] /pet/{petId}[39m
  [36m * [39m[36mdescription: Returns all pets from the system that the user has access to[39m
  [36m * [39m[36mparameters:[39m
  [36m * [39m[36m  - (path) petId=2* {Integer} The pet ID[39m
  [36m * [39m[36m  - (query) limit {Integer:int32} The number of resources to return[39m
  [36m */[39m
  [90mpublic String getPet(id) {[39m"
`;

exports[`CLI utils #swaggerInlineExample should support \`.js\` 1`] = `
"  [36m/*[39m
  [36m * [39m[36m@oas [get] /pet/{petId}[39m
  [36m * [39m[36mdescription: Returns all pets from the system that the user has access to[39m
  [36m * [39m[36mparameters:[39m
  [36m * [39m[36m  - (path) petId=2* {Integer} The pet ID[39m
  [36m * [39m[36m  - (query) limit {Integer:int32} The number of resources to return[39m
  [36m */[39m
  [90mroute.get(\\"/pet/:petId\\", pet.show);[39m"
`;

exports[`CLI utils #swaggerInlineExample should support \`.jsx\` 1`] = `
"  [36m/*[39m
  [36m * [39m[36m@oas [get] /pet/{petId}[39m
  [36m * [39m[36mdescription: Returns all pets from the system that the user has access to[39m
  [36m * [39m[36mparameters:[39m
  [36m * [39m[36m  - (path) petId=2* {Integer} The pet ID[39m
  [36m * [39m[36m  - (query) limit {Integer:int32} The number of resources to return[39m
  [36m */[39m
  [90mroute.get(\\"/pet/:petId\\", pet.show);[39m"
`;

exports[`CLI utils #swaggerInlineExample should support \`.php\` 1`] = `
"  [36m/*[39m
  [36m * [39m[36m@oas [get] /pet/{petId}[39m
  [36m * [39m[36mdescription: Returns all pets from the system that the user has access to[39m
  [36m * [39m[36mparameters:[39m
  [36m * [39m[36m  - (path) petId=2* {Integer} The pet ID[39m
  [36m * [39m[36m  - (query) limit {Integer:int32} The number of resources to return[39m
  [36m */[39m
  [90mfunction showPet($id) {[39m"
`;

exports[`CLI utils #swaggerInlineExample should support \`.py\` 1`] = `
"  [36m\\"\\"\\"[39m
  [36m@oas [get] /pet/{petId}[39m
  [36mdescription: Returns all pets from the system that the user has access to[39m
  [36mparameters:[39m
  [36m  - (path) petId=2* {Integer} The pet ID[39m
  [36m  - (query) limit {Integer:int32} The number of resources to return[39m
  [36m\\"\\"\\"[39m
  [90mdef getPet(id):[39m"
`;

exports[`CLI utils #swaggerInlineExample should support \`.rb\` 1`] = `
"  [36m=begin[39m
  [36m@oas [get] /pet/{petId}[39m
  [36mdescription: Returns all pets from the system that the user has access to[39m
  [36mparameters:[39m
  [36m  - (path) petId=2* {Integer} The pet ID[39m
  [36m  - (query) limit {Integer:int32} The number of resources to return[39m
  [36m=end[39m
  [90mget '/pet/:petId' do[39m"
`;

exports[`CLI utils #swaggerInlineExample should support \`.ts\` 1`] = `
"  [36m/*[39m
  [36m * [39m[36m@oas [get] /pet/{petId}[39m
  [36m * [39m[36mdescription: Returns all pets from the system that the user has access to[39m
  [36m * [39m[36mparameters:[39m
  [36m * [39m[36m  - (path) petId=2* {Integer} The pet ID[39m
  [36m * [39m[36m  - (query) limit {Integer:int32} The number of resources to return[39m
  [36m */[39m
  [90mroute.get(\\"/pet/:petId\\", pet.show);[39m"
`;
